src/test/kotlin/com/atlassian/performance/tools/report/api/judge/RelativeNonparametricPerformanceJudgeTest.kt

```diff
@@ -0,0 +1,54 @@
+package com.atlassian.performance.tools.report.api.judge
+
+import com.atlassian.performance.tools.jiraactions.api.EDIT_ISSUE
+import com.atlassian.performance.tools.report.api.ShiftedDistributionRegressionTest
+import com.atlassian.performance.tools.report.api.result.DurationData
+import com.atlassian.performance.tools.report.api.result.EdibleResult
+import com.atlassian.performance.tools.report.junit.FailedActionJunitReport
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.mockito.Mockito
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when` as whenever
+
+class RelativeNonparametricPerformanceJudgeTest {
+    private fun <T> any(type: Class<T>): T = Mockito.any<T>(type)
+
+    @Test
+    fun shouldProduceExpectedFailedActionReportMessage() {
+        // given
+        val actionToleranceRatio = 0.1f
+        val shiftedDistributionRegressionTest = mock(ShiftedDistributionRegressionTest::class.java).apply {
+            whenever(isExperimentRegressed(any(Double::class.java))).thenReturn(true)
+        }
+        val shiftedDistributionRegressionTestProvider = mock(ShiftedDistributionRegressionTestProvider::class.java).apply {
```

Comment:
`ShiftedDistributionRegressionTestProvider` is already very easy to mock via subclass, no need for Mockito.

Reply:
But it's literally an `open class`

Reply:
interfaces work better in such cases - no constructor problems